concordium-std-derive/src/lib.rs

```diff
@@ -1202,6 +1224,154 @@ fn schema_type_fields(fields: &syn::Fields) -> syn::Result<proc_macro2::TokenStr
     }
 }
 
+/// We reserve a number of error codes for custom errors, such as ParseError,
+/// that are provided by concordium-std. The error codes for user-defined error
+/// types will have this number as an offset.
+const RESERVED_ERROR_CODES: u32 = 100;
```

Comment:
I think it is going to be better if the "reserved" codes are not -1, -2, ... but that we start `i32::MIN, i32::MIN+1, ...`

This way user defined codes are going to be less scary, and more direct, and things that are kind of "system errors" will have weird looking codes.

Reply:
I think that's a good idea, will do.